### PR TITLE
Disable record-minmax-thread-safety-test

### PR DESCRIPTION
**Pull request by jinevening**
This PR disables record-minmax-thread-safety test to avoid gcc package bug.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: https://github.com/Samsung/ONE/issues/11202

### CODE REVIEW

File: compiler/record-minmax/CMakeLists.txt [hunk 1]
```diff
@@ -23,30 +23,35 @@ if(NOT ENABLE_TEST)
   return()
 endif(NOT ENABLE_TEST)
 
+###
```
Comment (seanshpark) [inline compiler/record-minmax/CMakeLists.txt:26]: plz split this to another PR

File: compiler/record-minmax-thread-safety-test/CMakeLists.txt [hunk 2]
```diff
@@ -2,6 +2,11 @@ if(NOT ENABLE_TEST)
     return()
 endif(NOT ENABLE_TEST)
 
+# Disable the test if record-minmax-for-thread-test does not exist
+if (NOT TARGET record-minmax-for-thread-test)
+    return()
```
Comment (seanshpark) [inline compiler/record-minmax-thread-safety-test/CMakeLists.txt:7]: optional)
can you add a message to show that `record-minmax-for-thread-test` is temporary disabled?

File: compiler/record-minmax/CMakeLists.txt [hunk 3]
```diff
@@ -23,30 +23,35 @@ if(NOT ENABLE_TEST)
...
-  target_link_libraries(record-minmax-for-thread-test luci_interpreter)
-  target_link_libraries(record-minmax-for-thread-test dio_hdf5)
-  target_link_libraries(record-minmax-for-thread-test vconone)
-  target_link_libraries(record-minmax-for-thread-test nncc_coverage)
-  target_link_libraries(record-minmax-for-thread-test luci_log)
-
-  target_compile_options(record-minmax-for-thread-test PUBLIC -fsanitize=thread)
-  target_link_libraries(record-minmax-for-thread-test -fsanitize=thread)
-endif("${CMAKE_SIZEOF_VOID_P}" STREQUAL "8")
+# if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "8")
```
Comment (seanshpark) [inline compiler/record-minmax/CMakeLists.txt:34]: Instead of adding `#` to this whole block,
can you revise this something like
```
if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "8" AND False)
```

Comment (hseok-oh) [reply compiler/record-minmax/CMakeLists.txt:34]: We can use `CMAKE_CXX_COMPILER_VERSION` to disable under gcc-10.

```cmake
if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "8" AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
  add_executable(record-minmax-for-thread-test ${DRIVER} ${SOURCES})
....

endif()

Comment (jinevening) [reply compiler/record-minmax/CMakeLists.txt:34]: @hseok-oh Thanks for the suggestion. I'd like to go with @seanshpark 's way, because we know `gcc-10` is ok, but not tested `gcc-11,12,13`. I think it would be safer to simply block the test for a while.

**Comment by jinevening** [comment]
Done.
